compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/InconsistentWarningsFix.scala

```diff
@@ -0,0 +1,244 @@
+// Copyright 2024 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see http://www.gnu.org/licenses/.
+
+package org.alephium.ralph.lsp.access.compiler
+
+import org.alephium.ralph._
+import org.alephium.protocol.vm._
+
+private object InconsistentWarningsFix {
+
+  /*
+   * The mutable state of `Ast.ContractWithState` is causing inconsistent warnings when compiling twice the same AST.
```

Comment:
If the AST does not change, why do you want to compile it twice instead of reusing the cached compiled objects?